Src/BootCamp.Chapter/Demo.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+
+    public class Demo
+    {
+        public event EventHandler OnKeyPressed;
+
+        private List<char> validChars = new List<char> { 'a', 'b', 'c', 'q' };
+
+        ContactsCenter contactsCenter = new ContactsCenter("./Input/MOCK_DATA.csv");
+
+        public char currentChar;
+
+        private bool _demoRunning = false;
+
+        public bool DemoRunning
+        {
+            get { return _demoRunning; }
+            set
+            {
+                if (_demoRunning != value)
+                {
+                    _demoRunning = value;
+                    Console.WriteLine(_demoRunning == true ? "Demo started." : "Demo Ended.");
+                }
+            }
+        }
+
+        public Demo()
+        {
+            StartDemo();
+        }
+
+        private void StartDemo()
+        {
+            PrintMenuChoice();
+
+            OnKeyPressed += RunDemoMenu;
+
+            do
+            {
+                if (GetKeyChar() == ' ')
+                {
+                    DemoRunning = true;
+                    OnKeyPressed?.Invoke(this, null);
+                }
+            } while (DemoRunning);
+        }
+
+        private void RunDemoMenu(object sender, EventArgs eventArgs)
+        {
+            OnKeyPressed -= RunDemoMenu;
```

Comment:
Pretty cool to see you try and switch behavior like this dynamically.

Reply:
Thanks!